lightning-background-processor/src/lib.rs

```diff
@@ -764,34 +758,44 @@ mod tests {
 	}
 
 	#[test]
-	fn test_not_handling_events_until_until_graph_sync_completion() {
-		let mut nodes = create_nodes(2, "test_not_handling_events_until_until_graph_sync_completion".to_string());
+	fn test_not_pruning_network_graph_until_graph_sync_completion() {
+		let mut nodes = create_nodes(2, "test_not_pruning_network_graph_until_graph_sync_completion".to_string());
 		let channel_value = 100000;
 		let data_dir = nodes[0].persister.get_data_dir();
 		let persister = Persister::new(data_dir.clone());
+		let network_graph = nodes[0].network_graph.clone();
+		let features = ChannelFeatures::empty();
+		network_graph.update_channel_from_partial_announcement(42, 53, features, nodes[0].node.get_our_node_id(), nodes[1].node.get_our_node_id());
+		let original_graph_description = network_graph.to_string();
+		assert!(original_graph_description.contains("42: features: 0000, node_one:"));
 
-		// Set up a background event handler for FundingGenerationReady events.
 		let (sender, receiver) = std::sync::mpsc::sync_channel(1);
 		let event_handler = move |event: &Event| {
 			sender.send(handle_funding_generation_ready!(event, channel_value)).unwrap();
 		};
 		let bg_processor = BackgroundProcessor::start(persister, event_handler, nodes[0].chain_monitor.clone(), nodes[0].node.clone(), nodes[0].net_graph_msg_handler.clone(), nodes[0].peer_manager.clone(), true, nodes[0].logger.clone());
 
-		// Open a channel and check that the FundingGenerationReady event was handled.
-		begin_open_channel!(nodes[0], nodes[1], channel_value);
-		let reception_result = receiver
-			.recv_timeout(Duration::from_secs(EVENT_DEADLINE));
-		assert!(reception_result.is_err());
+		for i in 0..EVENT_DEADLINE {
+			std::thread::sleep(Duration::from_secs(1));
```

Comment:
Hmm... we should avoid sleeps in tests if at all possible. I'd imagine you could have a persister that signals completion using `std::sync::mpsc::sync_channel` similar to the event handler above.

Reply:
Unfortunately, I can't really because the network pruning is not calling the event handler. All it does is call a method on the network graph without informing anybody. It's especially tricky because I wanna make sure that more time has passed than FIRST_NETWORK_PRUNE_TIMER.